compiler/compiler_test.go

```diff
@@ -0,0 +1,150 @@
+package compiler
+
+import (
+	"bytes"
+	"fmt"
+	"go/ast"
+	"go/build"
+	"go/parser"
+	"go/token"
+	"go/types"
+	"testing"
+
+	"github.com/sergi/go-diff/diffmatchpatch"
+	"golang.org/x/tools/go/loader"
+)
+
+func TestOrder(t *testing.T) {
+	fileA := `
+package foo
+
+var Avar = "a"
+
+type Atype struct{}
+
+func Afunc() int {
+	var varA = 1
+	var varB = 2
+	return varA+varB
+}
+`
+
+	fileB := `
+package foo
+
+var Bvar = "b"
+
+type Btype struct{}
+
+func Bfunc() int {
+	var varA = 1
+	var varB = 2
+	return varA+varB
+}
+`
+	files := []source{{"fileA.go", []byte(fileA)}, {"fileB.go", []byte(fileB)}}
+
+	compare(t, "foo", files, false)
+	compare(t, "foo", files, true)
+}
+
+func compare(t *testing.T, path string, sourceFiles []source, minify bool) {
+	outputNormal, err := compile(path, sourceFiles, minify)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// reverse the array
+	for i, j := 0, len(sourceFiles)-1; i < j; i, j = i+1, j-1 {
+		sourceFiles[i], sourceFiles[j] = sourceFiles[j], sourceFiles[i]
+	}
+
+	outputReversed, err := compile(path, sourceFiles, minify)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if string(outputNormal) != string(outputReversed) {
```

Comment:
We are already using https://github.com/google/go-cmp in other tests, let's stick to it. https://github.com/gopherjs/gopherjs/blob/be292294a6971bc1df34f570fc9d174322ab12bc/build/context_test.go#L51-L53 has an idiomatic example of using that package.